web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -1316,13 +1316,13 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
       return allottedHeight;
     }
 
-    const layers = this.layerGroup.layers;
+    const layers = this.layerGroup.spec.layer;
     let oskHeight = 0;
 
     // In case the keyboard's layers have differing row counts, we check them all for the maximum needed oskHeight.
     for (const layerID in layers) {
       const layer = layers[layerID];
-      let nRows = layer.rows.length;
+      let nRows = layer.row.length;
```

Comment:
Why this change?

Reply:
Took a bit to find it, but the reason is on line 1319.  It shifts from using the set of _pre-built_ OSK layers (which have a `.rows` property) to using the underlying touch-layout specification (which uses `.row`, not `.rows`) as the basis for our calculations here.

Reply:
Understood. A bit squidgy, but make sense!